diff -ur a/lib/handler/compress/brotli.cc b/lib/handler/compress/brotli.cc
--- a/lib/handler/compress/brotli.cc 01:03:13.000000000 +0000
+++ b/lib/handler/compress/brotli.cc 14:02:50.165728560 +0000
@@ -23,6 +23,18 @@
 #include <cstdlib>
 #include <vector>
 #include <zlib.h>
+
+#if defined(H2O_USE_LIBUV)
+# if H2O_USE_LIBUV
+#  include <uv.h>
+# endif
+#else
+# if H2O_USE_SELECT || H2O_USE_EPOLL || H2O_USE_KQUEUE
+# else
+#  include <uv.h>
+# endif
+#endif
+
 #include "h2o.h"
 #include "encode.h"
 
